beacon_node/network/src/beacon_processor/worker/sync_methods.rs

```diff
@@ -39,76 +39,101 @@ impl<T: BeaconChainTypes> Worker<T> {
     pub fn process_rpc_block(
         self,
         block: SignedBeaconBlock<T::EthSpec>,
-        result_tx: BlockResultSender<T::EthSpec>,
+        peer_id: PeerId,
+        process_type: BlockProcessType,
         reprocess_tx: mpsc::Sender<ReprocessQueueMessage<T>>,
         duplicate_cache: DuplicateCache,
     ) {
-        let block_root = block.canonical_root();
-        // Checks if the block is already being imported through another source
-        if let Some(handle) = duplicate_cache.check_and_insert(block_root) {
-            let slot = block.slot();
-            let block_result = self.chain.process_block(block);
+        // Check if the block is already being imported through another source
+        let root = block.canonical_root();
+        let handle = match duplicate_cache.check_and_insert(root) {
+            Some(handle) => handle,
+            None => {
+                return;
```

Comment:
I'm not sure if these cases are handled, but I would imagine that when we are requesting blocks to be processed, there is some state or mapping keeping track as to whether the blocks get processed successfully or fail, in parent lookup or single block lookup. Potentially, for this function its not the case, but if it is, I think we need to always emit a response indicating the block failed processing, to handle it by whoever requested it.
Here we just return.